lib/puppet/provider/gnupg_key/gnupg.rb

```diff
@@ -53,7 +53,11 @@ def add_key
   end
 
   def add_key_from_key_server
-    command = "gpg --keyserver #{resource[:key_server]} --recv-keys #{resource[:key_id]}"
+    if resource[:proxy].empty?
```

Comment:
Thanks for the work on this, minus the 1 note pertaining to ?empty ?nil on the resource it works great.